src/MemUtils_SYCL.cpp

```diff
@@ -0,0 +1,77 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   Implementation file for routines used to manage
+ *          memory for SYCL reductions and other operations.
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "RAJA/config.hpp"
+
+#if defined(RAJA_ENABLE_SYCL)
+
+#include "RAJA/policy/sycl/MemUtils_SYCL.hpp"
+
+
+namespace RAJA
+{
+
+namespace sycl
+{
+
+namespace detail
+{
+//
+/////////////////////////////////////////////////////////////////////////////
+//
+// Variables representing the state of execution.
+//
+/////////////////////////////////////////////////////////////////////////////
+//
+
+//! State of the host code globally
+syclInfo g_status;
+
+//! State of the host code in this thread
+syclInfo tl_status;
+#if defined(RAJA_ENABLE_OPENMP) && defined(_OPENMP)
+#pragma omp threadprivate(tl_status)
+#endif
```

Comment:
Will this link with the symbol in the header if openmp is enabled?

Reply:
Not correctly, is there a reason this needs to be `extern` rather than having functions that can retrieve or set it?